src/main/java/core/basesyntax/service/converter/DataConverterImpl.java

```
@@ -0,0 +1,32 @@
+package core.basesyntax.service.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String SPLIT = ",";
+    private static final int INDEX_TYPE = 0;
+    private static final int INDEX_FRUIT = 1;
+    private static final int INDEX_NUMBER = 2;
+
+    @Override
+    public List<FruitTransaction> convertAll(List<String> line) {
+        return line.stream()
+                .skip(1)
+                .map(this::lineSplit)
+                .map(f -> new FruitTransaction(
+                        Operation.getOperationType(f[INDEX_TYPE].toLowerCase()),
+                        f[INDEX_FRUIT], Integer.parseInt(f[INDEX_NUMBER])))
+                .collect(Collectors.toList());

```

Consider wrapping the entire conversion of a single line in a try-catch to catch NumberFormatException and IllegalArgumentException from `Operation.getOperationType` and then rethrow a RuntimeException that includes the original exception as cause and an informative message with the raw line content. The checklist emphasizes informative exception rethrowing for invalid data. Example guidance is in the checklist: "Don't ignore exceptions in try-catch blocks... rethrow a RuntimeException with an informative message and exception object."